daml-script/daml3/Daml/Script/Questions/Submit/Error.daml

```diff
@@ -121,34 +122,38 @@ data SubmitError
       truncatedErrorMessage : Text
 
--- TODO[SW]: Add more detail here
+instance Show AnyContractKey where
+  show (AnyContractKey _ templateTypeRep) = show templateTypeRep
+
 instance Show SubmitError where
   show err = case err of
     ContractNotFound { unknownContractIds, additionalDebuggingInfo } ->
       "Could not find contract with ID(s) " <> intercalate ", " (toList unknownContractIds) <> "."
       <> foldMap (\info -> " Additional debugging info: " <> show info) additionalDebuggingInfo
-    ContractKeyNotFound {} -> "Could not find contract with given key"
+    ContractKeyNotFound contractKey -> "Could not find contract with given key" <> show contractKey
     AuthorizationError errMessage -> "Authorization failure: " <> errMessage
-    DisclosedContractKeyHashingError contractId _ _ -> "Disclosed contract with ID " <> show contractId.contractId <> " used incorrect payload"
-    DuplicateContractKey {} -> "Attempted to create a contract with a contract key that is already in use"
-    InconsistentContractKey {} -> "Inconsistent contract key between daml and ledger"
-    UnhandledException {} -> "An Exception was thrown and not caught"
+    DisclosedContractKeyHashingError contractId contractKey contractKeyHash ->
+      "Disclosed contract with ID " <> show contractId <> " used incorrect payload - contract key: " <> show contractKey <> "; and contract key hash: " <> contractKeyHash
+    DuplicateContractKey contractKey -> "Attempted to create a contract with a contract key that is already in use - contract key: " <> show contractKey
+    InconsistentContractKey contractKey -> "Inconsistent contract key between daml and ledger - contract key: " <> show contractKey
+    UnhandledException None -> "An Exception was thrown and not caught"
+    UnhandledException (Some exn) -> "An Exception was thrown and not caught - message: " <> (message exn)
     UserError errMessage -> "User Error: " <> errMessage
     TemplatePreconditionViolated {} -> "Template precondition was violated"
-    CreateEmptyContractKeyMaintainers {} -> "Attempted to create a contract with empty key maintainers"
-    FetchEmptyContractKeyMaintainers {} -> "Attempted to fetch a contract with empty key maintainers"
-    WronglyTypedContract _ expected actual ->
-      "Excepted contract to be a \"" <> templateTypeRepToText expected <> "\" but it was actually a \"" <> templateTypeRepToText actual <> "\""
-    ContractDoesNotImplementInterface _ t i ->
-      "\"" <> templateTypeRepToText t <> "\" does not implement interface \"" <> templateTypeRepToText i <> "\""
-    ContractDoesNotImplementRequiringInterface _ t required requiring ->
-      "\"" <> templateTypeRepToText t <> "\" does not implement interface \"" <> templateTypeRepToText required <> "\" as required by \"" <> templateTypeRepToText requiring <> "\""
+    CreateEmptyContractKeyMaintainers _ -> "Attempted to create a contract with empty key maintainers"
```

Comment:
With `AnyTemplate` arguments, there's not a lot that we can do.

Reply:
Thats unfortunate, and ofc in a stable package so we can't do anything without adding a primative/question. I'm okay with this as is.